execute/observation.go

```diff
@@ -236,8 +241,14 @@ func (p *Plugin) getMessagesObservation(
 		return exectypes.Observation{}, fmt.Errorf("unable to observe costly messageObs %w", err)
 	}
 
+	hashes, err := exectypes.GetHashes(ctx, messageObs, p.msgHasher)
+	if err != nil {
+		return exectypes.Observation{}, fmt.Errorf("unable to get message hashes: %w", err)
+	}
+
 	observation.CommitReports = commitReportCache
 	observation.Messages = messageObs
+	observation.Hashes = hashes
```

Comment:
The hash is technically only required if the message data is removed, so you could have a final pass at the end of `truncateObservation` which removes redundant hashes to save even more space.

Reply:
good optimization going forward for sure 👍